src/registry.ts

```diff
@@ -67,12 +67,11 @@ export default class Registry implements RegistryInterface {
   public query(
     query: string | Node | Scope,
     scope: Scope = Scope.ANY,
-  ): Attributor | BlotConstructor | null {
+  ): RegistryDefinition | null {
     let match;
     if (typeof query === 'string') {
       match = this.types[query] || this.attributes[query];
-      // @ts-expect-error
-    } else if (query instanceof Text || query.nodeType === Node.TEXT_NODE) {
+    } else if (query instanceof Text) {
```

Comment:
Let's keep the `query.nodeType === Node.TEXT_NODE` check. It was introduced to fix an Edge browser issue: https://github.com/quilljs/parchment/commit/47385fb483ed87000547b998d21fa9411b00fcc9. Even though we don't support Edge 10 anymore, we should focus the scope of this PR to just typing.